### PR TITLE
[OB3] Docker resources for WSO2 Open Banking V3.0.0

### PR DESCRIPTION
**Pull request by AmilaSamith**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: dockerfiles/alpine/obam/Dockerfile [hunk 1]
```diff
@@ -44,51 +47,74 @@ ARG MOTD='printf "\n\
 ENV ENV=${USER_HOME}"/.ashrc"
 
 # create the non-root user and group and set MOTD login message
```
Comment (chirangaalwis) [inline dockerfiles/alpine/obam/Dockerfile]: @AmilaSamith do we require lines 49-50? As per my observation don't need them, please correct me if I am wrong.

If yes you can remove them.

Please check the other Dockerfiles as well.

**Review by chirangaalwis — CHANGES_REQUESTED** [review]

File: dockerfiles/alpine/obam/Dockerfile (hunk 1 above)
Comment (AmilaSamith) [reply dockerfiles/alpine/obam/Dockerfile]: Yes we need them @chirangaalwis as we need to override MOTD of base image

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -23,21 +23,21 @@ The issues page on GitHub is for tracking bugs and feature requests. When posing
 
 If you like to contribute with a bug fix or a new feature, start by posting an issue and discussing the best way to implement it.
 
-Unlike most projects, development for this repository is carried out on the `2.0.0` branch. This is because the `master` branch contains the latest stable release of the project.
-The code in `2.0.0` is merged to the `master` branch after a final review and a round of testing.
+Unlike most projects, development for this repository is carried out on the `3.0.0` branch. This is because the `master` branch contains the latest stable release of the project.
```
Comment (chirangaalwis) [inline CONTRIBUTING.md:26]: @AmilaSamith this has to be `3.0.x` if I am not mistaken.

**Review by chirangaalwis — CHANGES_REQUESTED** [review]

File: dockerfiles/alpine/obam/README.md [hunk 3]
```diff
@@ -41,51 +44,30 @@ git clone https://github.com/wso2/docker-open-banking.git
 ## How to update configurations
 
 Configurations would lie on the Docker host machine and they can be volume mounted to the container. <br>
-As an example, steps required to change the port offset using `carbon.xml` is as follows:
+As an example, steps required to change the port offset using `deployment.toml` is as follows:
 
 ##### 1. Stop the API Manager container if it's already running.
 
-In WSO2 Open Banking API Manager 2.0.0 product distribution, `carbon.xml` configuration file <br>
+In WSO2 Open Banking API Manager 2.0.0 product distribution, `deployment.toml` configuration file <br>
```
Comment (chirangaalwis) [inline dockerfiles/alpine/obam/README.md]:
```suggestion
In WSO2 Open Banking API Manager 3.0.0 product distribution, `deployment.toml` configuration file <br>
```

File: dockerfiles/alpine/obiam/Dockerfile [hunk 4]
```diff
@@ -46,48 +50,93 @@ ARG MOTD='printf "\n\
...
+# configure datasources
+    && sed -i -e 's|DB_APIMGT_URL|jdbc:mysql://'${DB_HOST}':3306/'${DB_APIMGT}'?autoReconnect=true\&amp;useSSL=false|g' ${DEPLOYMENT_TOML_FILE} \
+    && sed -i -e 's|DB_GOV_URL|jdbc:mysql://'${DB_HOST}':3306/'${DB_GOV}'?autoReconnect=true\&amp;useSSL=false|g' ${DEPLOYMENT_TOML_FILE} \
+    && sed -i -e 's|DB_IS_CONFIG_URL|jdbc:mysql://'${DB_HOST}':3306/'${DB_IS_CONFIG}'?autoReconnect=true\&amp;useSSL=false|g' ${DEPLOYMENT_TOML_FILE} \
+    && sed -i -e 's|DB_USER_STORE_URL|jdbc:mysql://'${DB_HOST}':3306/'${DB_USER_STORE}'?autoReconnect=true\&amp;useSSL=false|g' ${DEPLOYMENT_TOML_FILE} \
+    && sed -i -e 's|DB_OB_STORE_URL|jdbc:mysql://'${DB_HOST}':3306/'${DB_OPEN_BANKING_STORE}'?autoReconnect=true\&amp;useSSL=false|g' ${DEPLOYMENT_TOML_FILE} \
+    && sed -i -e 's|DB_USER|'${DB_USER}'|g' ${DEPLOYMENT_TOML_FILE} \
+    && sed -i -e 's|DB_PASS|'${DB_PASS}'|g' ${DEPLOYMENT_TOML_FILE} \
+    && sed -i -e 's|DB_DRIVER|'${DB_DRIVER}'|g' ${DEPLOYMENT_TOML_FILE} \
+# Copy deployment.toml file to repository/conf
```
Comment (chirangaalwis) [inline dockerfiles/alpine/obiam/Dockerfile]:
```suggestion
# copy deployment.toml file to original product pack location
```

File: dockerfiles/alpine/obiam/Dockerfile [hunk 5]
```diff
@@ -46,48 +50,93 @@ ARG MOTD='printf "\n\
...
+    && cp ${DEPLOYMENT_TOML_FILE} ${WSO2_SERVER_HOME}/repository/conf/ \
+    && rm ${DEPLOYMENT_TOML_FILE}
 
-# install required packages
-RUN apk add --no-cache netcat-openbsd
+# copy the WSO2 OBIAM accelerator artifacts on top of WSO2 IS base product
+# Remove old open banking artifacts from base product
```
Comment (chirangaalwis) [inline dockerfiles/alpine/obiam/Dockerfile]:
```suggestion
# remove old open banking artifacts from base product
```

File: dockerfiles/alpine/obiam/Dockerfile [hunk 6]
```diff
@@ -46,48 +50,93 @@ ARG MOTD='printf "\n\
...
+RUN \
+    find ${WSO2_SERVER_HOME}/repository/components/dropins -name "com.wso2.openbanking.*" -exec rm -rf {} \; \
+    && find ${WSO2_SERVER_HOME}/repository/components/lib -name "com.wso2.openbanking.*" -exec rm -rf {} \; \
+# Copying open banking artifacts
```
Comment (chirangaalwis) [inline dockerfiles/alpine/obiam/Dockerfile]:
```suggestion
# copy open banking artifacts
```

File: dockerfiles/ubuntu/obam/Dockerfile [hunk 7]
```diff
@@ -42,54 +47,88 @@ This Docker container comprises of The WSO2 Open Banking Solution, with its late
...
-    && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
-        libxml2-utils \
-        netcat \
-        unzip \
-        wget \
-    && rm -rf /var/lib/apt/lists/*
-# add the WSO2 product distribution to user's home directory
+    find ${WSO2_SERVER_HOME}/repository/components/dropins -name "com.wso2.openbanking.*" -exec rm -rf {} \; \
+    && find ${WSO2_SERVER_HOME}/repository/components/lib -name "com.wso2.openbanking.*" -exec rm -rf {} \; \
+# Copying open banking artifacts
```
Comment (chirangaalwis) [inline dockerfiles/ubuntu/obam/Dockerfile]:
```suggestion
# copy open banking artifacts
```

File: dockerfiles/ubuntu/obam/README.md [hunk 8]
```diff
@@ -41,51 +45,30 @@ git clone https://github.com/wso2/docker-open-banking.git
 ## How to update configurations
 
 Configurations would lie on the Docker host machine and they can be volume mounted to the container. <br>
-As an example, steps required to change the port offset using `carbon.xml` is as follows:
+As an example, steps required to change the port offset using `deployment.toml` is as follows:
 
 ##### 1. Stop the API Manager container if it's already running.
 
-In WSO2 Open Banking API Manager 2.0.0 product distribution, `carbon.xml` configuration file <br>
+In WSO2 Open Banking API Manager 2.0.0 product distribution, `deployment.toml` configuration file <br>
```
Comment (chirangaalwis) [inline dockerfiles/ubuntu/obam/README.md]:
```suggestion
In WSO2 Open Banking API Manager 3.0.0 product distribution, `deployment.toml` configuration file <br>
```

File: dockerfiles/ubuntu/obiam/Dockerfile [hunk 9]
```diff
@@ -45,53 +49,95 @@ This Docker container comprises of The WSO2 Open Banking Solution, with its late
...
+# configure datasources
+    && sed -i -e 's|DB_APIMGT_URL|jdbc:mysql://'${DB_HOST}':3306/'${DB_APIMGT}'?autoReconnect=true\&amp;useSSL=false|g' ${DEPLOYMENT_TOML_FILE} \
+    && sed -i -e 's|DB_GOV_URL|jdbc:mysql://'${DB_HOST}':3306/'${DB_GOV}'?autoReconnect=true\&amp;useSSL=false|g' ${DEPLOYMENT_TOML_FILE} \
+    && sed -i -e 's|DB_IS_CONFIG_URL|jdbc:mysql://'${DB_HOST}':3306/'${DB_IS_CONFIG}'?autoReconnect=true\&amp;useSSL=false|g' ${DEPLOYMENT_TOML_FILE} \
+    && sed -i -e 's|DB_USER_STORE_URL|jdbc:mysql://'${DB_HOST}':3306/'${DB_USER_STORE}'?autoReconnect=true\&amp;useSSL=false|g' ${DEPLOYMENT_TOML_FILE} \
+    && sed -i -e 's|DB_OB_STORE_URL|jdbc:mysql://'${DB_HOST}':3306/'${DB_OPEN_BANKING_STORE}'?autoReconnect=true\&amp;useSSL=false|g' ${DEPLOYMENT_TOML_FILE} \
+    && sed -i -e 's|DB_USER|'${DB_USER}'|g' ${DEPLOYMENT_TOML_FILE} \
+    && sed -i -e 's|DB_PASS|'${DB_PASS}'|g' ${DEPLOYMENT_TOML_FILE} \
+    && sed -i -e 's|DB_DRIVER|'${DB_DRIVER}'|g' ${DEPLOYMENT_TOML_FILE} \
+# Copy deployment.toml file to repository/conf
```
Comment (chirangaalwis) [inline dockerfiles/ubuntu/obiam/Dockerfile]:
```suggestion
# copy deployment.toml file to original product pack location
```

File: dockerfiles/ubuntu/obiam/Dockerfile [hunk 10]
```diff
@@ -45,53 +49,95 @@ This Docker container comprises of The WSO2 Open Banking Solution, with its late
...
+    && cp ${DEPLOYMENT_TOML_FILE} ${WSO2_SERVER_HOME}/repository/conf/ \
+    && rm ${DEPLOYMENT_TOML_FILE}
 
-# add the WSO2 product distribution to user's home directory
+# copy the WSO2 OBIAM accelerator artifacts on top of WSO2 IS base product
+# Remove old open banking artifacts from base product
```
Comment (chirangaalwis) [inline dockerfiles/ubuntu/obiam/Dockerfile]:
```suggestion
# remove old open banking artifacts from base product
```

File: dockerfiles/ubuntu/obiam/Dockerfile [hunk 11]
```diff
@@ -45,53 +49,95 @@ This Docker container comprises of The WSO2 Open Banking Solution, with its late
...
 RUN \
-    wget -O ${WSO2_SERVER}.zip "${WSO2_SERVER_DIST_URL}" \
-    && unzip -d ${USER_HOME} ${WSO2_SERVER}.zip \
-    && chown wso2carbon:wso2 -R ${WSO2_SERVER_HOME} \
-    && rm -f ${WSO2_SERVER}.zip
+    find ${WSO2_SERVER_HOME}/repository/components/dropins -name "com.wso2.openbanking.*" -exec rm -rf {} \; \
+    && find ${WSO2_SERVER_HOME}/repository/components/lib -name "com.wso2.openbanking.*" -exec rm -rf {} \; \
+# Copying open banking artifacts
```
Comment (chirangaalwis) [inline dockerfiles/ubuntu/obiam/Dockerfile]:
```suggestion
# copy open banking artifacts
```

**Review by chirangaalwis — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (AmilaSamith) [reply CONTRIBUTING.md:26]: Yes it is I will make the change

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -23,21 +23,21 @@ The issues page on GitHub is for tracking bugs and feature requests. When posing
...
+2. Create a topic branch from the `3.0.x` branch.
 3. Make commits in logical units.
 4. Before you send out the pull request, sync your forked repository with a remote repository. This makes your pull request simple and clear.
 
 ```bash
 git clone https://github.com/<user>/docker-open-banking.git
 git remote add upstream https://github.com/wso2/docker-open-banking.git
 git fetch upstream
-git checkout -b <topic-branch> upstream/2.0.0
+git checkout -b <topic-branch> upstream/3.0.0
```
Comment (chirangaalwis) [inline CONTRIBUTING.md:40]:
```suggestion
git checkout -b <topic-branch> upstream/3.0.x
```

File: dockerfiles/ubuntu/obiam/README.md [hunk 13]
```diff
@@ -1,13 +1,19 @@
-# Dockerfile for WSO2 Open Banking Identity & Access Management Module #
+# Dockerfile for WSO2 Open Banking Identity & Access Management Accelerator Module #
```
Comment (chirangaalwis) [inline dockerfiles/ubuntu/obiam/README.md]:
```suggestion
# Dockerfile for WSO2 Open Banking Identity & Access Management Accelerator Module
```

**Review by chirangaalwis — CHANGES_REQUESTED** [review]

**Review by chirangaalwis — APPROVED** [review]

File: docker-compose/obam-with-obiam/mysql/scripts/permission.sql [hunk 14]
```diff
@@ -0,0 +1 @@
+ALTER USER 'root'@'localhost' IDENTIFIED WITH mysql_native_password BY '9900';
```
Comment (chirangaalwis) [inline docker-compose/obam-with-obiam/mysql/scripts/permission.sql:1]: @AmilaSamith please add an EOF line.

**Review by chirangaalwis — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 15]
```diff
@@ -1,37 +1,20 @@
...
+- Solution is based on WSO2 APIM 4.0.0 and WSO2 IS 5.11.0  
+- Upgrade profile based dockerfiles to use docker.wso2.com/wso2is:5.11.0.0 and docker.wso2.com/wso2is:5.11.0.0 as the base images for OBAM and OBIAM respectively.
+- Previous XML based configurations are carried out using TOML based configurations. Please refer the documentations for detailed mapping of configuration changes 
+  - [Configuring Identity Server](https://ob.docs.wso2.com/en/latest/install-and-setup/configuring-identity-server-for-ob/)
+  - [Configuring API Manager](https://ob.docs.wso2.com/en/latest/install-and-setup/configuring-api-manager-for-ob/)
+- For detailed information of the WSO2 Open Banking 3.0.0 product release, please see the GitHub release notes [Releases](https://github.com/wso2-enterprise/financial-open-banking/releases)
 
-For detailed information on the tasks carried out during this release, please see the GitHub milestone [v2.0.0.1](https://github.com/wso2/docker-open-banking/milestone/11)
+For detailed information on the tasks carried out during this release, please see the GitHub milestone [v3.0.0.1](https://github.com/wso2/docker-open-banking/milestone/11)
```
Comment (chirangaalwis) [inline CHANGELOG.md:18]: @AmilaSamith if you haven't created a GitHub Milestone with the issues covered in this release, let's remove this line. Currently, it points to an older milestone.

File: CHANGELOG.md [hunk 16]
```diff
@@ -1,37 +1,20 @@
...
-  as Identity & Access Management Module version `2.0.0` and WSO2 Open Banking Business Intelligence Dashboard and Worker profiles `2.0.0`
-- Upgrade profile based dockerfiles to use Adopt OpenJDK `jdk8u272-b10` as the base image
-- Docker Compose resources for OBAM, OBIAM with OBBI deployment patterns based on UK and Berlin specifications
-- Docker Compose resources for OBAM with OBIAM deployment patterns based on UK and AU specifications  
-- Solution is based on WSO2 APIM 3.1.0 and WSO2 ISKM 5.10.0  
-- Previous XML based configurations are carried out using TOML based configurations. Please refer the documentation for detailed mapping of configuration changes [Map Configurations](https://docs.wso2.com/display/OB200/Map+XML+Configurations+to+TOML+Configurations)
-- For detailed information of the WSO2 Open Banking 2.0.0 product release, please see the GitHub release notes [Releases](https://github.com/wso2-enterprise/financial-open-banking/releases)
+- Alpine and Ubuntu based Docker resources for WSO2 Open Banking API Manager Accelerator Module version `3.0.0` and WSO2 Open Banking Identity & Access Management Accelerator Module version `3.0.0`
```
Comment (chirangaalwis) [inline CHANGELOG.md:9]: @AmilaSamith if you have created an issue for this task, please refer it here.

**Review by chirangaalwis — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 15 above)
Comment (AmilaSamith) [reply CHANGELOG.md]: Removed line related to milestone

File: CHANGELOG.md (hunk 16 above)
Comment (AmilaSamith) [reply CHANGELOG.md]: added references for the issues

**Review by chirangaalwis — APPROVED** [review]

**Pull request merged**